### ISSUE TITLE
pay_all_debts.groovy:38-40: Add test for pay_all_depts...

### ISSUE THREAD

**Issue by 0pdd** [zerocracy/farm#903]
The puzzle `696-65091594` from #696 has to be resolved:

https://github.com/zerocracy/farm/blob/d06f16c3ef055e3dd5827fb4f044d7083f745d53/src/main/resources/com/zerocracy/stk/pmo/debts/pay_all_debts.groovy#L38-L40

The puzzle was created by g4s8 on 02-Apr-18. 

Estimate: 30 minutes, role: DEV.

If you have any technical questions, don't ask me, submit new tickets instead. The task will be "done" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this issue

**Comment by 0crat** [comment]
@0pdd[/z](https://www.0crat.com/u/0pdd) this project will fix the problem faster if you donate a few dollars to it; just [click here](https://www.0crat.com/contrib/C3NDPUA8L) and pay via Stripe, it's very fast, convenient and appreciated; thanks a lot!

**Comment by 0crat** [comment]
Job [#903](https://github.com/zerocracy/farm/issues/903) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
The job [#903](https://github.com/zerocracy/farm/issues/903) assigned to @g4s8[/z](https://www.0crat.com/u/g4s8), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1561009545); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by g4s8** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/zerocracy/farm/issues/903#issuecomment-400267623))

@g4s8 The user @g4s8[/z](https://www.0crat.com/u/g4s8) resigned from [#903](https://github.com/zerocracy/farm/issues/903), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by 0crat** [comment]
The job [#903](https://github.com/zerocracy/farm/issues/903) assigned to @krzyk[/z](https://www.0crat.com/u/krzyk), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1791553513); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by krzyk** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/zerocracy/farm/issues/903#issuecomment-402106274))

@krzyk The user @krzyk[/z](https://www.0crat.com/u/krzyk) resigned from [#903](https://github.com/zerocracy/farm/issues/903), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

**Comment by 0crat** [comment]
The job [#903](https://github.com/zerocracy/farm/issues/903) assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1880861056); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by carlosmiranda** [comment]
@0crat wait for completion of #1069, it's a blocker.

**Comment by 0crat** [comment]
> @0crat wait for completion of #1069, it's a blocker. ([here](https://github.com/zerocracy/farm/issues/903#issuecomment-403368971))

@carlosmiranda The impediment for [#903](https://github.com/zerocracy/farm/issues/903) was registered successfully by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by g4s8** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/zerocracy/farm/issues/903#issuecomment-472956233))

@g4s8 The user @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) resigned from [#903](https://github.com/zerocracy/farm/issues/903), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
The job [#903](https://github.com/zerocracy/farm/issues/903) assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/3cefe2ca-fbc7-43d9-9fb9-c93439f01300); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by victornoel** [comment]
@g4s8 how do you choose the prefix in the bundle names? Do I put anything unique or is there some rule there?

**Comment by victornoel** [comment]
@0crat waiting for ARC answer

**Comment by g4s8** [comment]
@victornoel yes, it's a workaround for `BundleTest`, it should be any unique string

**Comment by 0crat** [comment]
> @0crat waiting for ARC answer ([here](https://github.com/zerocracy/farm/issues/903#issuecomment-544239803))

@victornoel The impediment for [#903](https://github.com/zerocracy/farm/issues/903) was registered successfully by @victornoel[/z](https://www.0crat.com/u/victornoel)

**Comment by victornoel** [comment]
@g4s8 thanks, and another question: the todo is coming from #696 which doesn't apply to zld payment method (zld is always paid directly). Since zld is the unique payment method, what should we be testing for as a starter? Just that debts are paid?

**Comment by g4s8** [comment]
@victornoel we have `FkBank` implementation which is called by `Payroll` when testing. You can use this class for test.

**Comment by victornoel** [comment]
@0crat refuse

The current architecture is too complex for me to handle this :)

**Comment by 0crat** [comment]
> @0crat refuse The current architecture is too complex for me to handle this :) ([here](https://github.com/zerocracy/farm/issues/903#issuecomment-586593451))

@victornoel The user @victornoel[/z](https://www.0crat.com/u/victornoel) resigned from [#903](https://github.com/zerocracy/farm/issues/903), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged (job refused in 3007 hours), see [§6](http://www.zerocracy.com/policy.html#6): -20 point(s) just awarded to @victornoel[/z](https://www.0crat.com/u/victornoel)

**Comment by 0crat** [comment]
The job [#903](https://github.com/zerocracy/farm/issues/903) assigned to @loro2[/z](https://www.0crat.com/u/loro2), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/2b721abc-b6d5-4610-b04d-511f5a436777); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by loro2** [comment]
@refuse

No time atm, going for vacation

**Comment by loro2** [comment]
@0crat refuse
